examples/voting/ballot.v.py

```diff
@@ -32,61 +43,88 @@ def __init__(_proposalNames: bytes32[2]):
 # May only be called by `chairperson`.
 def give_right_to_vote(voter: address):
-    # Throws if sender is not chairpers
+    # Throws if sender is not chairperson
     assert msg.sender == self.chairperson
     # Throws if voter has already voted
     assert not self.voters[voter].voted
-    # Throws if voters voting weight isn't 0 
+    # Throws if voter's voting weight isn't 0
     assert self.voters[voter].weight == 0
     self.voters[voter].weight = 1
     self.voter_count += 1
 
+# Used by `delegate`. Can be called by anyone.
+def forward_weight(delegate_with_weight_to_forward: address):
+    assert self.delegated(delegate_with_weight_to_forward)
+    # Throw if there is nothing to do:
+    assert self.voters[delegate_with_weight_to_forward].weight > 0
+
+    target = self.voters[delegate_with_weight_to_forward].delegate
+    for i in range(4):
+        if self.delegated(target):
+            target = self.voters[target].delegate
+            # The following effectively detects cycles of length <= 5.
+            # BUT cycles aren't actually problematic for correctness;
+            # they just result in spoiled votes.
+            # So, in the production version, this should instead be
+            # the responsibility of the contract's client, and this
+            # check should be removed.
+            assert target != delegate_with_weight_to_forward
+        else:
+            # weight will be moved to someone who directly voted or
+            # hasn't voted
+            break
+
+    weight_to_forward = self.voters[delegate_with_weight_to_forward].weight
+    self.voters[delegate_with_weight_to_forward].weight = 0
+    self.voters[target].weight += weight_to_forward
+
+    if self.directly_voted(target):
+        self.proposals[self.voters[target].vote].vote_count += weight_to_forward
+        self.voters[target].weight = 0
+
+    # To reiterate: if target is also a delegate, this function will need
+    # to be called again.
+
 # Delegate your vote to the voter `to`.
-def delegate(_to: address):
-    to = _to
+def delegate(to: address):
     # Throws if sender has already voted
-    assert not self.voters[msg.sender].voted 
-    # Throws if sender tries to delegate their vote to themselves
-    assert not msg.sender == to
-    # loop can delegate votes up to the current voter count
-    for i in range(self.voter_count, self.voter_count+1):
-        if self.voters[to].delegate:
-        # Because there are not while loops, use recursion to forward the delegation
-        # self.delegate(self.voters[to].delegate)
-            assert self.voters[to].delegate != msg.sender 
-            to = self.voters[to].delegate
+    assert not self.voters[msg.sender].voted
+    # Throws if sender tries to delegate their vote to themselves or to
+    # the default address value of 0x0000000000000000000000000000000000000000
+    # (the latter might not be problematic, but I don't want to think about it)
+    assert to != msg.sender and not not to
```

Comment:
Is there a reason you're using `not not ` instead of just checking for a `truthy` value?